app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,41 @@
 class PostsController < ApplicationController
-  def index
+  def find_user
     @user = User.find(params[:user_id])
   end
 
+  def index
+    find_user
+  end
+
   def show
-    @post = User.find(params[:user_id]).posts.find(params[:id])
+    @post = find_user.posts.find(params[:id])
```

Comment:
- [ ] <b>REQUIRED:</b> Kindly make sure to use `current_user` here as well before you merge this PR.